### PR TITLE
Migrate to go modules

### PR DESCRIPTION
**Pull request by PatrickKvartsaniy**
Hello:)

I just tried to migrate centrifugo to the go modules. Can you review it, please?

### CODE REVIEW

**Comment by FZambia** [comment]
Thanks a lot @PatrickKvartsaniy ! Will take a look.

BTW - have you had a chance to think on release workflow - will it continue to work with goreleaser and [packages.sh](https://github.com/centrifugal/centrifugo/blob/master/misc/scripts/package.sh) will properly build binary to be used in Linux distributions? Or this was completely out of scope? I am asking because this must be tested anyway and any information regarding to this will help.

File: Makefile [hunk 1]
```diff
@@ -45,3 +47,7 @@ docs-deploy: docs-image
 
 docs-image:
 	docker build -t $(DOC_IMAGE) -f docs/Dockerfile docs/
+
+deps:
+	go mod download
```
Comment (FZambia) [inline Makefile]: I think `go mod vendor` is enough here?

**Comment by PatrickKvartsaniy** [comment]
@FZambia Hello again:) I made some changes and update a `Travis` config

**Comment by PatrickKvartsaniy** [comment]
> Thanks a lot @PatrickKvartsaniy ! Will take a look.
> 
> BTW - have you had a chance to think on release workflow - will it continue to work with goreleaser and [packages.sh](https://github.com/centrifugal/centrifugo/blob/master/misc/scripts/package.sh) will properly build binary to be used in Linux distributions? Or this was completely out of scope? I am asking because this must be tested anyway and any information regarding to this will help.

Travis config has been updated. Everything in the make file also updated for work with modules. Nothing should be affected, but I'm not able to test it more than on CI. In case I missed something deeply in the config, just add -mod=vendor to each `go build` `go test`

**Comment by FZambia** [comment]
Thanks!

**Pull request merged**